server/src/db/chunk.rs

```diff
@@ -438,10 +438,35 @@ impl QueryChunk for DbChunk {
     /// However, since we current sorted data based on their cardinality (see compute_sort_key),
     /// 2 different chunks may be sorted on different order of key columns.
     fn is_sorted_on_pk(&self) -> bool {
-        match &self.state {
-            State::MutableBuffer { .. } => false,
-            State::ReadBuffer { .. } => false,
-            State::ParquetFile { .. } => false,
+        match self.schema().sort_key() {
+            Some(_sort_key) => return true,
+            _ => return false,
+        }
+    }
+
+    /// Returns the sort key of the chunk if any
+    fn sort_key(&self) -> Option<SortKey<'_>> {
+        // Try1
+        // let schema  = self.schema();
+        // schema.sort_key()
+
+        // try 2
+        // let schema_cloned = self.schema().as_ref().clone();
+        // let schema = Arc::new(schema_cloned);
+        // let sort_key = schema.sort_key();
+        // sort_key
+
+        // try3 : still compile error
+        let sort_key = self.schema().sort_key();
```

Comment:
```suggestion
        let sort_key = self.meta.schema.sort_key();
```

The problem with this code as currently formulated is is that `schema()` returns some new `Arc` so the `SortKey` would have a reference to that arc. By using `self.meta.schema` directly, the returned SortKey now has a reference to `&self` so rust can validate nothing changes `&self`